test/unittests/Connection_test.cpp

```diff
@@ -0,0 +1,134 @@
+}
+
+TEST_F(ConnectionCreationTest, DefaultConstructorBusOpenSystemThrow)
+{
+    EXPECT_CALL(*mock_, sd_bus_open_system(_)).WillOnce(DoAll(SetArgPointee<0>(STUB_), Return(-1)));
+    ASSERT_THROW(sdbus::internal::Connection(BusType::eSystem, std::move(mock_)), sdbus::Error);
+}
+
+TEST_F(ConnectionCreationTest, DefaultConstructorBusOpenSessionThrow)
+{
+    EXPECT_CALL(*mock_, sd_bus_open_user(_)).WillOnce(DoAll(SetArgPointee<0>(STUB_), Return(-1)));
+    ASSERT_THROW(sdbus::internal::Connection(BusType::eSession, std::move(mock_)), sdbus::Error);
+}
+
+TEST_F(ConnectionCreationTest, DefaultConstructorFlushSystemThrow)
+{
+    EXPECT_CALL(*mock_, sd_bus_open_system(_)).WillOnce(DoAll(SetArgPointee<0>(STUB_), Return(1)));
+    EXPECT_CALL(*mock_, sd_bus_flush(_)).WillOnce(Return(-1));
+    EXPECT_CALL(*mock_, sd_bus_flush_close_unref(_)).Times(1);
+    ASSERT_THROW(sdbus::internal::Connection(BusType::eSystem, std::move(mock_)), sdbus::Error);
+}
+
+TEST_F(ConnectionCreationTest, DefaultConstructorFlushSessionThrow)
+{
+    EXPECT_CALL(*mock_, sd_bus_open_user(_)).WillOnce(DoAll(SetArgPointee<0>(STUB_), Return(1)));
+    EXPECT_CALL(*mock_, sd_bus_flush(_)).WillOnce(Return(-1));
+    EXPECT_CALL(*mock_, sd_bus_flush_close_unref(_)).Times(1);
+    ASSERT_THROW(sdbus::internal::Connection(BusType::eSession, std::move(mock_)), sdbus::Error);
+}
+
+
+
+class ConnectionRequestTest : public ::testing::TestWithParam<BusType> {
+protected:
+    ConnectionRequestTest() = default;
+    void SetUp() override
+    {
+        switch (GetParam())
+        {
+            case BusType::eSystem:
+                EXPECT_CALL(*mock_, sd_bus_open_system(_)).WillOnce(DoAll(SetArgPointee<0>(STUB_), Return(1)));
+                break;
+            case BusType::eSession:
+                EXPECT_CALL(*mock_, sd_bus_open_user(_)).WillOnce(DoAll(SetArgPointee<0>(STUB_), Return(1)));
+                break;
+            default:
+                break;
+        }
+        EXPECT_CALL(*mock_, sd_bus_flush(_)).Times(1);
+        EXPECT_CALL(*mock_, sd_bus_flush_close_unref(_)).Times(1);
+    }
+
+    std::unique_ptr<Dbus_interface_mock> mock_ { std::make_unique<Dbus_interface_mock>(Dbus_interface_mock()) };
+    sd_bus* STUB_ { reinterpret_cast<sd_bus*>(1) };
+};
+
+TEST_P(ConnectionRequestTest, RequestNoThrow)
+{
+    EXPECT_CALL(*mock_, sd_bus_request_name(_, _, _)).WillOnce(Return(1));
+
+    auto conn_ = sdbus::internal::Connection(GetParam(), std::move(mock_)) ;
+    ASSERT_NO_THROW(conn_.requestName(""));
+}
+
+TEST_P(ConnectionRequestTest, RequestThrow)
+{
+    EXPECT_CALL(*mock_, sd_bus_request_name(_, _, _)).WillOnce(Return(-1));
+
+    auto conn_ = sdbus::internal::Connection(GetParam(), std::move(mock_)) ;
+    ASSERT_THROW(conn_.requestName(""), sdbus::Error);
+}
+
+INSTANTIATE_TEST_SUITE_P(Request, ConnectionRequestTest, ::testing::Values(BusType::eSystem, BusType::eSession));
```

Comment:
Please add a blank line at the end of file. This helps a lot when printing the file on a console for example, with command prompt continuing on next line, not just after the last line of printed file.